### PR TITLE
feat: C509 craft a tbs certificate and sign the certificate

### PR DESCRIPTION
**Pull request by bkioshn**
# Description

Craft a TBS certificate for C509 Certificate and sign the certificate

## Description of Changes
- Create a TBS certificate
- Create signing function using Ed25519
- Fix encode and decode bug
- Add test cases

## Please confirm the following checks

* [x] My code follows the style guidelines of this project
* [x] I have performed a self-review of my code
* [x] I have commented my code, particularly in hard-to-understand areas
* [x] I have made corresponding changes to the documentation
* [x] My changes generate no new warnings
* [x] I have added tests that prove my fix is effective or that my feature works
* [x] New and existing unit tests pass locally with my changes
* [x] Any dependent changes have been merged and published in downstream module

### CODE REVIEW

**Review by stevenj — APPROVED** [review]
LGTM

File: catalyst-gateway-crates/c509-certificate/src/c509_name/mod.rs [hunk 1]
```diff
@@ -270,27 +273,34 @@ fn create_rdn_with_cn_attr(text: String) -> NameValue {
...
+    pub(crate) fn name_cn_text() -> (Name, String) {
         let mut attr = Attribute::new(oid!(2.5.4 .3));
         attr.add_value(AttributeValue::Text("RFC test CA".to_string()));
         let mut rdn = RelativeDistinguishedName::new();
         rdn.add_attr(attr);
 
-        let name = Name::new(NameValue::RelativeDistinguishedName(rdn));
+        (
+            Name::new(NameValue::RelativeDistinguishedName(rdn)),
+            "6b5246432074657374204341".to_string(),
```
Comment (Mr-Leshiy) [inline catalyst-gateway-crates/c509-certificate/src/c509_name/mod.rs:290]: Not clear what this string means. and how it was generated.

File: catalyst-gateway-crates/c509-certificate/src/c509_name/mod.rs [hunk 2]
```diff
@@ -344,23 +354,30 @@ mod test_name {
...
+    // A.1.  Example RFC 7925 profiled X.509 Certificate
+    pub(crate) fn name_cn_eui_mac() -> (Name, String) {
+        let mut attr = Attribute::new(oid!(2.5.4 .3));
+        attr.add_value(AttributeValue::Text("01-23-45-FF-FE-67-89-AB".to_string()));
+        let mut rdn = RelativeDistinguishedName::new();
+        rdn.add_attr(attr);
+
+        (
+            Name::new(NameValue::RelativeDistinguishedName(rdn)),
+            "47010123456789ab".to_string(),
```
Comment (Mr-Leshiy) [inline catalyst-gateway-crates/c509-certificate/src/c509_name/mod.rs:367]: same

File: catalyst-gateway-crates/c509-certificate/src/c509_name/mod.rs [hunk 3]
```diff
@@ -468,14 +481,21 @@ mod test_name {
         rdn.add_attr(attr4);
         rdn.add_attr(attr5);
 
-        let name = Name::new(NameValue::RelativeDistinguishedName(rdn));
+        (
+            Name::new(NameValue::RelativeDistinguishedName(rdn)),
+            "8a0462555306624341086b4578616d706c6520496e63096d63657274696669636174696f6e016a3830322e314152204341".to_string(),
```
Comment (Mr-Leshiy) [inline catalyst-gateway-crates/c509-certificate/src/c509_name/mod.rs:486]: same

**Pull request merged**